packages/devextreme/js/__internal/core/utils/promise.ts

```diff
@@ -0,0 +1,22 @@
+/* eslint-disable @typescript-eslint/init-declarations */
+/* eslint-disable spellcheck/spell-checker */
+
+/**
+ * This function is substitution for `Promise.withResolvers`
+ * and should be replaced by native one once possible
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/withResolvers}
+ */
+export function createPromise<T>(): {
+  promise: Promise<T>;
+  resolve: (value: T) => void;
+  reject: (value: T) => void;
+} {
+  let resolve!: (value: T) => void;
+  let reject!: (value: T) => void;
+  const promise = new Promise<T>((res, rej) => {
+    resolve = res;
+    reject = rej;
+  });
+
+  return { promise, resolve, reject };
```

Comment:
Minor comment.

In my opinion, it's better to return the array instead of the object
It's very similar in functionality, but let's compare these two blocks of code
Array:
```typescript
const [myPromise1, myResolve1, myReject1] = createPromise<void>();
const [myPromise2, myResolve2, myReject2] = createPromise<void>();
```
Object:
```typescript
const { promise: myPromise1, resolve: myResolve1, reject: myReject1 } = createPromise<void>();
const { promise: myPromise2, resolve: myResolve2, reject: myReject2 } = createPromise<void> ();
```

But objects are more comfortable when we want to deconstruct only one property:
Array:
```typescript
const [,  , myReject1] = createPromise<void>();
const [,  , myReject2] = createPromise<void>();
```
Object:
```typescript
const { reject: myReject1 } = createPromise<void>();
const { reject: myReject2 } = createPromise<void> ();
```

But in my opinion arrays still better (but it's just my opinion)

Reply:
I decided to use objects so that my helper had similar API as future native `Promise.withResolvers`